subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/gmm/GradleModuleMetadataAvailableAtIntegrationTest.groovy

```diff
@@ -0,0 +1,221 @@
+            'org:moduleA:1.0' {
+                variants(["api", "runtime"]) {
+                    availableAt("../../external/1.0/external-1.0.module", "org", "external", "1.0")
+                }
+            }
+            'org:external:1.0' {
+                dependsOn('org:do_not_reach:1.0')
+            }
+        }
+
+        buildFile << """
+            dependencies {
+                conf("org:moduleA:1.0") {
+                    transitive = false
+                }
+            }
+        """
+
+        when:
+        repositoryInteractions {
+            'org:moduleA:1.0' {
+                expectGetMetadata()
+            }
+            'org:external:1.0' {
+                expectResolve()
+            }
+        }
+        run ":checkDeps"
+
+        then:
+        resolve.expectGraph {
+            root(":", ":test:") {
+                module("org:moduleA:1.0") {
+                    noArtifacts()
+                    module("org:external:1.0")
+                }
+            }
+        }
+    }
+
+    def "resolves available-at variant even if configuration is not transitive"() {
+        given:
+        repository {
+            'org:moduleA:1.0' {
+                variants(["api", "runtime"]) {
+                    availableAt("../../external/1.0/external-1.0.module", "org", "external", "1.0")
+                }
+            }
+            'org:external:1.0' {
+                dependsOn('org:do_not_reach:1.0')
+            }
+        }
+
+        buildFile << """
+            dependencies {
+                conf("org:moduleA:1.0")
+            }
+            configurations.conf.transitive = false
+        """
+
+        when:
+        repositoryInteractions {
+            'org:moduleA:1.0' {
+                expectGetMetadata()
+            }
+            'org:external:1.0' {
+                expectResolve()
+            }
+        }
+        run ":checkDeps"
+
+        then:
+        resolve.expectGraph {
+            root(":", ":test:") {
+                module("org:moduleA:1.0") {
+                    noArtifacts()
+                    module("org:external:1.0")
+                }
+            }
+        }
+    }
+
+    @Issue("https://github.com/gradle/gradle/issues/14017")
+    @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "maven")
+    def "resolves available-at variant even if transitive dependencies are excluded"() {
+        given:
+        mavenHttpRepo.module('org', 'moduleB', '1.0')
+            .dependsOn(mavenHttpRepo.module("org", "moduleA", "1.0"), exclusions: [[group: '*', module: '*']])
+            .publish() // no Gradle Module metadata for this one because we don't support excludes on dependencies
+        repository {
+            'org:moduleA:1.0' {
+                variants(["api", "runtime"]) {
+                    availableAt("../../external/1.0/external-1.0.module", "org", "external", "1.0")
+                }
+            }
+            'org:external:1.0'()
```

Comment:
Could this one also depend on `'org:do_not_reach:1.0'`?

Reply:
Good catch indeed adding this highlights a bug in the implementation :cry: